pkgs/development/beam-modules/default.nix

```diff
@@ -37,9 +37,23 @@ let
         buildMix = callPackage ./build-mix.nix {};
 
         # BEAM-based languages.
-        elixir = if versionAtLeast (lib.getVersion erlang) "18"
-                 then callPackage ../interpreters/elixir { debugInfo = true; }
-                 else throw "Elixir requires at least Erlang/OTP R18.";
+        elixir = elixir-1_4;
+
+        elixir-1_5 = lib.callElixir ../interpreters/elixir/1.5.nix "18" {
+                       inherit rebar erlang;
+                       debugInfo = true;
+                     };
+
+        elixir-1_4 = lib.callElixir ../interpreters/elixir/1.4.nix "18" {
```

Comment:
Yes, I mean adding an argument and asserting it [here](https://github.com/NixOS/nixpkgs/pull/26716/files#diff-3885c55c990cee7fd6033e2f70fbe734R5) then it's contained in the erlang builder. And if elixir16 would only work with R19 it could specify it [here](https://github.com/NixOS/nixpkgs/pull/26716/files#diff-3272a642671bdb3d9acba450e3e56845R4) along with it's version, etc.